File: client/src/api/rules/RuleCatalogResponse.ts
import type { RuleDefinitionDto } from './RuleDefinitionDto'
import type { RuleEntityDto } from './RuleEntityDto'
import type { RulesetDto } from './RulesetDto'
import type { RuleSourceVersionDto } from './RuleSourceVersionDto'
import type { SourceBookDto } from './SourceBookDto'

export type RuleCatalogResponse = {
  rulesets: RulesetDto[]
  sourceBooks: SourceBookDto[]
  sourceVersions: RuleSourceVersionDto[]
  entities: RuleEntityDto[]
  definitions: RuleDefinitionDto[]
}

File: server/src/GenesysForge.Contracts/Rules/RuleCatalogResponse.cs
namespace GenesysForge.Contracts.Rules;

public sealed record RuleCatalogResponse(
    IReadOnlyCollection<RulesetDto> Rulesets,
    IReadOnlyCollection<SourceBookDto> SourceBooks,
    IReadOnlyCollection<RuleSourceVersionDto> SourceVersions,
    IReadOnlyCollection<RuleEntityDto> Entities,
    IReadOnlyCollection<RuleDefinitionDto> Definitions);
